src/Adapter/MSTestAdapter.PlatformServices/ObjectModel/UnitTestElement.cs

```diff
@@ -221,19 +226,25 @@ private Guid GenerateSerializedDataStrategyTestId(string testFullName)
             // In case path contains invalid characters.
         }
 
-        idProvider.AppendString(fileNameOrFilePath);
-        idProvider.AppendString(testFullName);
+        // TODO: Pool?
+        StringBuilder builder = new();
```

Comment:
This implementation seems to be doing different things than the test idProvider / xxhasher. 

Those two "append" string, but the string is not really appended (we don't consume more and more memory), instead we take the current hash and transform it to be the hash + the string, getting a new hash. This is more memory efficient and (based on my measurements in the past) faster, than creating the string, and then hashing it in 1 go.

Reply:
I added it anyways.

Reply:
Thanks for testing that!